packages/typescript-eslint/src/index.ts

```diff
@@ -16,6 +16,8 @@ import type {
 import { config } from './config-helper';
 import { getTSConfigRootDirFromStack } from './getTSConfigRootDirFromStack';
 
+export type { TSESLint } from '@typescript-eslint/utils';
```

Comment:
From https://github.com/typescript-eslint/typescript-eslint/issues/10848#issuecomment-2767600291 & https://github.com/typescript-eslint/typescript-eslint/issues/10848#issuecomment-2789340850:

> I think we should just export `FlatConfig` and keep it limited to that.

> Accepting PRs on just exporting `FlatConfig`. If more use cases come up we can consider them individually. 🚀

I asked for clarification in https://github.com/typescript-eslint/typescript-eslint/issues/10848#issuecomment-3347070366 on whether there should be more.